homeassistant/components/foscam/entity.py

```diff
@@ -0,0 +1,30 @@
+"""Component providing basic support for Foscam IP cameras."""
+from __future__ import annotations
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import DOMAIN
+from .coordinator import FoscamCoordinator
+
+
+class FoscamEntity(CoordinatorEntity[FoscamCoordinator]):
+    """Base entity for Foscam camera."""
+
+    def __init__(
+        self,
+        coordinator: FoscamCoordinator,
+        config_entry: ConfigEntry,
```

Comment:
So now we cut down on config entry usage. Might as well only pass the entry_id instead of the full entry :)